app/src/main/java/com/duckduckgo/app/browser/omnibar/OmnibarLayoutViewModel.kt

```diff
@@ -293,46 +294,51 @@ class OmnibarLayoutViewModel @Inject constructor(
     }
 
     fun onViewModeChanged(viewMode: ViewMode) {
+        val currentViewMode = _viewState.value.viewMode
         Timber.d("Omnibar: onViewModeChanged $viewMode")
-        when (viewMode) {
-            is CustomTab -> {
-                _viewState.update {
-                    it.copy(
-                        viewMode = viewMode,
-                        showClearButton = false,
-                        showVoiceSearch = false,
-                        showBrowserMenu = true,
-                        showTabsMenu = false,
-                        showFireIcon = false,
-                    )
+        if (currentViewMode is CustomTab) {
+            Timber.d("Omnibar: custom tab mode enabled, sending updates there")
```

Comment:
As mentioned on MM, I'm concerned if this will cause problems long term when we introduce leadingIconState, but I'm fine for now

Reply:
Agree that this is not ideal. I’m writing a proper proposal to clean up the custom tabs mode.